### DISCUSSION TITLE
Rule.Inclusion Option

### DISCUSSION THREAD

**Discussion by paulmcclbg** [Azure/PSRule.Rules.Azure#2676] [Q&A] [answered]
Does the rule include option support wild cards? e.g. say I only want to include all the APIM rules for execution. Do I have to list each one or is there a better way to achieve this ?

i.e. - 'PSRule.Rules.Azure\Azure.APIM.*' ? - (didn't appear to work)

**Comment by BernieWhite** [comment, marked as answer]
@paulmcclbg If you set `Rule.Include` to `Azure.APIM.*` then it should work. But it wasn't intended to be used like this and there is some limitations.

- `Rule.Include` will only support one entry.
- The wildcard will match the short name only not the full module\rule name.

Wildcards were intended to support usage via the `-Name` parameter of `Invoke/Assert/Get-PSRule` PowerShell cmdlets.

If you only want to run `Azure.APIM.*` rules this option should work.

---

Generally, running a subset of rules is supported by a baseline but they don't address your request specifically, because they have the same limitation. However you could generate a baseline by using this script as an example: https://github.com/Azure/PSRule.Rules.Azure/blob/main/scripts/security-baseline.ps1

**Reply by paulmcclbg** [reply]
@BernieWhite yes I ran in to the limitations you described. I have decided to go down the baseline route. Thanks.
